BungieFrameWork/BFW_Source/BFW_FileManager/Platform_Linux/BFW_FileManager_Linux.c

```diff
@@ -0,0 +1,1058 @@
+/*
+	FILE:	BFW_FileManager_Linux.c
+*/
+
+#include "BFW.h"
+#include "BFW_FileManager.h"
+#include "BFW_AppUtilities.h"
+
+#include <stdio.h>
+#include <string.h>
+
+//#include <iostream.h>
+#include <dirent.h>
+#include <errno.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+// resides in FileManager_Common.c but no public prototype
+UUtError BFrFile_MapClassic(BFtFileRef	*inFileRef,
+	UUtUns32			inOffset,
+	BFtFile				**outFile,
+	void				**outPtr,
+	UUtUns32			*outSize);
+
+
+#if 0
+struct BFtFileMapping
+{
+	BFtFile	*file;
+	void	*ptr;
+};
+#else
+struct BFtFileMapping
+{
+	char	name[BFcMaxPathLength];
+	UUtBool	mmapped;
+	BFtFile	*file;
+	void	*ptr;
+};
+#endif
+
+static void
+BFiConvertWin2Unix(
+	const char*	inOrigName,
+	char*		outMungedName)
+{
+	const char*	src;
+	char*		dst;
+	
+	src = inOrigName;
+	dst = outMungedName;
+	
+	while(*src != 0)
+	{
+		if(*src == '\\')
+		{
+			*dst++ = '/';
+			
+			src++;
+		}
+		else
+		{
+			*dst++ = *src;
+			
+			src++;
+		}
+	}
+	
+	*dst = 0;
+}
+
+static UUtBool
+BFiFileRef_PathNameExists(
+	const char*	inPath)
+{
+	return access(inPath, F_OK) == 0;
+}
+
+static UUtError
+BFiCheckFileName(
+	const char		*inFileName)
+{
+	const char		*name;
+	
+	name = inFileName;
+	
+	/*
```

Comment:
Let's add a comment about this function meant for checking for illegal paths on Windows,
which is a nop on Linux.
That way, we can remove the dead comment.